### ISSUE TITLE
Edit Feature or Better Sketch Origin

### ISSUE THREAD

**Issue by jbienz** [AlfMikael/snap-generator#5]
I've noticed there's not an easy way to go back and edit a cantilever feature. It _is_ possible to edit the sketch, but I've noticed that the sketch origin isn't placed at the mid-point even if I have the cantilever centered on the joint origin. Therefore, if I try to edit the extrusion length after the fact, it will no longer be centered on the joint origin. Thanks!

**Comment by AlfMikael** [comment]
Hey. The feature is not intended to be used that way. The intent is that if you need a different cantilever, you simply open the addin again and recreate it. The reason for this is that it is that it would be laboursome to implement together with the feature where you can pick the y-location as the origin point. The code would get far more messy and convoluted as well, so I don't see myself changing that anytime soon.

**Comment by jbienz** [comment]
Instead of doing an extrude of 100% from a far plane, would it be possible to a have the sketch placed at the center of the origin point (on the y direction) and extrude 50% on both sides? That would allow the extrude length to change or even be bound to a parameter without becoming uncentered over the origin.

**Comment by jbienz** [comment]
FYI, I'm asking because once I create a cantilever, I'm mirroring it on multiple axis. So, if I have to delete and recreate it to adjust a parameter, I also have to fix up the missing source errors for the mirrors. Thanks again.

**Comment by AlfMikael** [comment]
Yes, this is possible, and there existed an earlier implementation (before github) that worked this way. I left that behind because you don't always want to extrude from the middle. Sometimes you're extruding from one side, for example from an edge. This way simplifies the implementation to a simple offset, instead of using another extrude function. However, I do believe it is possible to change it in the way that you'd want, and I will admit I have been a little annoyed that I have to redo my mirrors whenever I want to modify the cantilever. We'll, see, it's food for thought :)

**Comment by jbienz** [comment]
Yeah, I understand. That does make sense. It's unfortunate that F360 doesn't have more of a concept of a plugin "feature object". At least not one that's easy to implement. I know this is more work, but maybe like how you have an X position and Y position you could have a Z position? And that would set the point where the sketch plane is anchored?

Anyway, it's an amazing tool. I left a comment on your YouTube video but I think it got spam canned. If you want to setup a method for donating I'm happy to give something. I really do appreciate this feature.

 Thanks for considering.

**Comment by AlfMikael** [comment]
Thanks for the kind words! If I start producing more content, then maybe I'll set up some donation feature. Very cool knowing that someone appreciates my work that much :).

And yes, as mentioned, it is possible to work around the issue with positioning, so that you can change the extrusion distance of the cantilever, while at the same time maintaining the center or edge positioning. In regards to a "feature object", Fusion actually is attempting to implement this [as we speak]( https://help.autodesk.com/view/fusion360/ENU/?guid=GUID-1C076040-8906-4D09-8649-81E90F431B0D). However, I have not looked closer into it, because it is still under development, and I don't want to have the code break because of their changes.
